### ISSUE TITLE
TraceID's not being logged as part of standard requests

### ISSUE THREAD

**Issue by proffalken** [traefik/traefik#8599]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I'm trying to get Traefik traces into Tempo to be visualised by Grafana, however in order to search for them I need the trace id to be in the log output so I can look for them via Loki.

I've configured tracing with the following setup:

```
[tracing]
  serviceName = "traefik"
  [tracing.zipkin]
    httpEndpoint = "https://zipkin/api/v2/spans"
```

Configured logging with the following:

```
[log]
  level = "DEBUG"
  filePath = "/var/log/traefik/traefik.log"
  format = "json"

[accessLog]
  filePath = "/var/log/traefik/access.log"
  format = "json"
```
I would expect this to add the trace ID to each log event in the access log so that when I process each request I can see the trace ID

### What did you see instead?

If I enable DEBUG logging then the trace ID's turn up in `/var/log/traefik/traefik.log` but not in the access log, if I disable DEBUG logging then I don't see the trace ID's at all.

### What version of Traefik are you using?

```
traefik version
Version:      dev
Codename:     cheddar
Go version:   go1.17.1
Built:        I don't remember exactly
OS/Arch:      linux/amd64
```
Installed via `pikaur` on Arch linux, Traefik UI says `dev`, pikaur says `community/traefik 2.5.3-1 [installed]`

### What is your environment & configuration?

```yaml
[entrypoints]
  [entrypoints.http]
    address = ":80"
  [entrypoints.https]
    address = ":443"
    [entryPoints.https.http.tls]
      certResolver = "le"
      [[entryPoints.https.http.tls.domains]]
          main = "*.service.my.tld"

[http.serversTransports.mytransport]
  insecureSkipVerify = true

[providers]
  [providers.consulCatalog]
    prefix = "traefik"
    requireConsistent = true
    exposedByDefault = true
    defaultRule = "Host(`{{ .Name }}.service.my.tld`)"
    [providers.consulCatalog.endpoint]
      address = "http://x.x.x.x:8500"
      scheme = "http"

[api]
  dashboard = true
  insecure = true
  debug = true

[metrics]
  [metrics.prometheus]
    addEntryPointsLabels = true
    addRoutersLabels = true
    addServicesLabels = true


[log]
  level = "DEBUG"
  filePath = "/var/log/traefik/traefik.log"
  format = "json"

[accessLog]
  filePath = "/var/log/traefik/access.log"
  format = "json"

[tracing]
  serviceName = "traefik"
  [tracing.zipkin]
    httpEndpoint = "https://otlpzipkin.service.my.tld/api/v2/spans"

# Enable ACME (Let's Encrypt): automatic SSL.
[certificatesResolvers.le.acme]
  email = "xxxx@xxxx.xxx"
  storage = "/etc/traefik.d/acme.json"
  [certificatesResolvers.le.acme.dnsChallenge]
    provider = "digitalocean"
    delayBeforeCheck = 10
    resolvers = ["1.1.1.1:53", "8.8.8.8:53"]

```

Add more configuration information here.

### If applicable, please paste the log output in DEBUG level

access.log
```json
{
   "ClientAddr" : "x.x.x.x:51538",
   "ClientHost" : "x.x.x.x",
   "ClientPort" : "51538",
   "ClientUsername" : "-",
   "DownstreamContentSize" : 0,
   "DownstreamStatus" : 202,
   "Duration" : 3240599,
   "OriginContentSize" : 0,
   "OriginDuration" : 2951927,
   "OriginStatus" : 202,
   "Overhead" : 288672,
   "RequestAddr" : "otlpzipkin.service.my.tld",
   "RequestContentSize" : 2480,
   "RequestCount" : 1047,
   "RequestHost" : "otlpzipkin.service.my.tld",
   "RequestMethod" : "POST",
   "RequestPath" : "/api/v2/spans",
   "RequestPort" : "-",
   "RequestProtocol" : "HTTP/2.0",
   "RequestScheme" : "https",
   "RetryAttempts" : 0,
   "RouterName" : "otlpzipkin@consulcatalog",
   "ServiceAddr" : "x.x.x.x:26107",
   "ServiceName" : "otlpzipkin@consulcatalog",
   "ServiceURL" : {
      "ForceQuery" : false,
      "Fragment" : "",
      "Host" : "x.x.x.x:26107",
      "Opaque" : "",
      "Path" : "",
      "RawFragment" : "",
      "RawPath" : "",
      "RawQuery" : "",
      "Scheme" : "http",
      "User" : null
   },
   "StartLocal" : "2021-11-30T14:40:07.588885557Z",
   "StartUTC" : "2021-11-30T14:40:07.588885557Z",
   "TLSCipher" : "TLS_AES_128_GCM_SHA256",
   "TLSVersion" : "1.3",
   "entryPointName" : "https",
   "level" : "info",
   "msg" : "",
   "time" : "2021-11-30T14:40:07Z"
}
```

traefik.log
```json
{
   "Request" : "{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/metrics\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\",\"RawFragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"application/openmetrics-text; version=0.0.1,text/plain;version=0.0.4;q=0.5,*/*;q=0.1\"],\"Accept-Encoding\":[\"gzip\"],\"User-Agent\":[\"Prometheus/2.31.1\"],\"X-B3-Parentspanid\":[\"73fc1ad6a47b0000\"],\"X-B3-Sampled\":[\"1\"],\"X-B3-Spanid\":[\"3323bfd4425966cc\"],\"X-B3-Traceid\":[\"03700aea388c32cb73fc1ad6a47b0000\"],\"X-Forwarded-Host\":[\"mqtt2prom.service.my.tld:443\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"hypervisor01.servers.my.tld\"],\"X-Prometheus-Scrape-Timeout-Seconds\":[\"10\"],\"X-Real-Ip\":[\"172.17.0.12\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"mqtt2prom.service.my.tld:443\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"172.17.0.12:50948\",\"RequestURI\":\"/metrics\",\"TLS\":null}",
   "level" : "debug",
   "msg" : "vulcand/oxy/roundrobin/rr: completed ServeHttp on request",
}
```

**Comment by kawhicurry** [comment]
I'm doing the same thing : `I'm trying to get Traefik traces into Tempo to be visualised by Grafana`. The `trace_id` should be a part of `accesslog` instead of the log of traefik itself.

**Comment by proffalken** [comment]
FWIW, this is still an issue in V3 AFAICT - Traces are enabled, but the trace ID's are *not* in the access logs in either CLF or JSON formats.

If it's a simple fix, then I might be able to raise a PR, but I know *nothing* about GOLANG and don't really have time to learn it right now sadly :(

**Comment by schewara** [comment]
Same problem here

As the icing on the cake, the most preferable solution would be if the logs (mainly thinking structured JSON here) would be in line with OpenTelemetry's [Logs Data Model](https://opentelemetry.io/docs/specs/otel/logs/data-model/) and [Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/).

Switching over to the OTel Log implementation would then be super easy in the future, once the functionality is available in the Go Instrumentation library.

**Comment by schewara** [comment]
Had another stab at it with the latest v3.1 and everything switched over to opentelemetry, it is still the same state.

Based on other suggestions in other places, I also tried to add the `traceparent`, `tracestate` and `baggage` headers to the log, but they seem to be not set, at least in my setup.

Here my config 
```yaml
accessLog:
  format: json
  bufferingSize: 100
  fields:
    names:
      StartLocal: drop
    headers:
      names:
        Referer: keep
        User-Agent: keep
        # W3C Trace-Context
        traceparent: keep
        tracestate: keep
        baggage: keep
```

**Comment by chewrocca** [comment]
@schewara i think this PR is going to help https://github.com/traefik/traefik/pull/10921

**Comment by nmengin** [comment]
Closed by #10921
